java/src/com/ibm/streamsx/topology/messaging/kafka/ConsumerConnector.java

```diff
@@ -0,0 +1,198 @@
+            throw new IllegalArgumentException("threadsPerTopic");
+
+        Map<String, Object> params = new HashMap<>();
+        params.put("topic", topics);
+        params.put("threadsPerTopic", threadsPerTopic);
+        if (!config.isEmpty())
+            params.put("kafkaProperty", Util.toKafkaProperty(config));
+        
+        // workaround streamsx.messaging issue #107
+        params.put("propertiesFile", PROP_FILE_PARAM);
+        addPropertiesFile();
+
+        // see streamsx.messaging issue #109
+        TStream<Message> rcvdMsgs;
+        boolean confirmedMultiTopicConsumerWorks = false;
+        if (topics.length==1 || confirmedMultiTopicConsumerWorks) {
+            SPLStream rawKafka = SPL.invokeSource(
+                            te,
+                            "com.ibm.streamsx.messaging.kafka::KafkaConsumer",
+                            params,
+                            KafkaSchemas.KAFKA);
+
+            rcvdMsgs = toMessageStream(rawKafka);
+        }
+        else {
+            // fake it with N kafkaConsumers
+            List<TStream<Message>> list = new ArrayList<TStream<Message>>(topics.length);
+            for (String topic : topics) {
+                Map<String,Object> myParams = new HashMap<>(params);
+                myParams.put("topic", topic);
+
+                SPLStream rawKafka = SPL.invokeSource(
+                        te,
+                        "com.ibm.streamsx.messaging.kafka::KafkaConsumer",
+                        myParams,
+                        KafkaSchemas.KAFKA);
+                
+                list.add(toMessageStream(rawKafka));
+            }
+            TStream<Message> kafka = list.remove(0);
+            
+            rcvdMsgs = kafka.union(new HashSet<>(list));
+        }
+        return rcvdMsgs;
+    }
+
+    /**
+     * Convert an {@link SPLStream} with schema {@link KafkaSchemas.KAFKA}
+     * to a TStream&lt;{@link Message}>.
+     * The returned stream will contain a {@code Message} tuple for
+     * each tuple on {@code stream}.
+     * A runtime error will occur if the schema of {@code stream} doesn't
+     * have the attributes defined by {@code KafkaSchemas.KAFKA}.
+     * @param stream Stream to be converted to a TStream&lt;Message>.
+     * @return Stream of {@code Message} tuples from {@code stream}.
+     */
+    private static TStream<Message> toMessageStream(SPLStream stream) {
+        return stream.convert(new Function<Tuple, Message>() {
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public Message apply(Tuple tuple) {
+                return new SimpleMessage(fromSplValue(tuple.getString("message")),
+                                    fromSplValue(tuple.getString("key")),
+                                    fromSplValue(tuple.getString("topic")));
+            }
+            
+            private String fromSplValue(String s) {
+                // SPL doesn't allow null value.  For our use,
```

Comment:
I think what ever the decision is made as to how to handle empty values, it should be clearly documented in the visible javadoc.